pkg/controller/rollout/trafficrouting/trafficroutingutil.go

```diff
@@ -16,12 +16,17 @@ limitations under the License.
 
 package trafficrouting
 
+import "context"
+
 // Controller common function across all TrafficRouting implementation
 type Controller interface {
-	// SetRoutes set canary ingress desired weight
-	SetRoutes(desiredWeight int32) error
-	// Verify check if canary ingress has been set desired weight
-	Verify(desiredWeight int32) (bool, error)
-	// Finalise will do some cleanup work, such as delete canary ingress
-	Finalise() error
+	// Validate will validate the traffic routing resource,
+	Validate(ctx context.Context) (bool, error)
+	// SetWeight set canary desired weight. desiredWeight[0,100]
+	SetWeight(ctx context.Context, desiredWeight int32) error
```

Comment:
plz consider sticking the original interface func name since they're more generic 
SetWeight -> SetRoutes
Verify -> VerifyRoutes

Validate sounds too generic, consider rename to more meaningful func name such as ensureRoutesExist or initialize